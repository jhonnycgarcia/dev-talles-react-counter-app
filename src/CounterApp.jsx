import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const [counter, setCount] = useState(value);

    const increment = () => {
        // setCount(counter + 1)
        setCount((prevCount) => prevCount + 1)
    }

    const decrement = () => {
        setCount((prevCount) => prevCount - 1)
    }

    const reset = () => {
        setCount(value)
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2> {counter} </h2>
            <button aria-label='btn-add' onClick={increment}>+1</button>
            <button aria-label='btn-sub' onClick={decrement}>-1</button>
            <button aria-label='btn-reset' onClick={reset}>Reset</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
};

// CounterApp.defaultProps = {
//     value: 0,
// }
