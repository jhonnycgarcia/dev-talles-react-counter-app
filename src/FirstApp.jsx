
import PropTypes from 'prop-types';

const getResult = () => {
  return 'Hola'
}

export const FirstApp = ({ name, title, subtitle }) => {

  return (
    <>
      <h1>{getResult() + ' ' + title}</h1>
      <div>{subtitle}</div>
      <div>{name}</div>
    </>
  )
}

FirstApp.propTypes = {
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  name: 'Pedro',
  subtitle: 'Este es el subtítulo por defecto',
  title: 'Título por defecto',
}