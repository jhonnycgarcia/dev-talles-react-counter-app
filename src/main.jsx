import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
// import { App } from './App'
import { FirstApp } from './FirstApp'

import './style.css'

createRoot(document.getElementById('root')).render(
    <StrictMode>
        {/* <App /> */}
        <FirstApp title='Pedro' subtitle='El primer app'/>
    </StrictMode>,
)