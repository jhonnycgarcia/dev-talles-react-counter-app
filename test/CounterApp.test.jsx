import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";


describe('Pruebas en CounterApp.jsx', () => {

    const counterValue = 100;

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<CounterApp value={ counterValue } />);
        expect( container ).toMatchSnapshot();
    });

    test('debe de mostrar el valor incial', () => {
        render(<CounterApp value={ counterValue } />);
        expect( screen.getByRole('heading', { level: 2 }) ).toBeTruthy();
        expect( screen.getByRole('heading', { level: 2 }).textContent ).toContain( counterValue.toString() );
    });

    test('debe incrementar el valor con el botón "+1"', () => {
        const result = (counterValue + 1).toString();
        render(<CounterApp value={ counterValue } />);
        fireEvent.click( screen.getByText('+1') );
        expect( screen.getByText('+1') ).toBeTruthy();
        expect( screen.getByText(result) ).toBeTruthy();
        expect( screen.getByRole('heading', { level: 2 }).textContent ).toContain( result );
    });

    test('debe decrementar el valor con el botón "-1"', () => {
        const result = (counterValue - 1).toString();
        render(<CounterApp value={ counterValue } />);
        fireEvent.click( screen.getByText('-1') );
        expect( screen.getByText('-1') ).toBeTruthy();
        expect( screen.getByText(result) ).toBeTruthy();
        expect( screen.getByRole('heading', { level: 2 }).textContent ).toContain( result );
    });

    test('debe resetear el valor con el botón "Reset"', () => {
        const result = '0';
        render(<CounterApp value={ 0 } />);
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );

        fireEvent.click( screen.getByRole('button', { name: 'btn-reset' }) );

        expect( screen.getByText('Reset') ).toBeTruthy();
        // screen.debug();
        expect( screen.getByText(result) ).toBeTruthy();
        expect( screen.getByRole('heading', { level: 2 }).textContent ).toContain( result );
    });

});