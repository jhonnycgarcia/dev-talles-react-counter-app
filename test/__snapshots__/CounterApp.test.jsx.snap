// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp.jsx debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button
    aria-label="btn-add"
  >
    +1
  </button>
  <button
    aria-label="btn-sub"
  >
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
